src/FormItem.tsx

```diff
@@ -18,25 +19,34 @@ export const FormItem = <TFieldValues extends FieldValues = FieldValues>({
   control,
   name,
   help,
+  valuePropName,
   ...props
 }: FormItemProps<TFieldValues>) => {
-  const { field, fieldState } = useController({ name, control });
-
-  const handleNormalize: AntdFormItemProps['normalize'] = (value) => {
-    field.onChange(value);
-    return value;
-  };
+  const { fieldState } = useController({ name, control });
 
   return (
-    <AntdForm.Item
-      {...props}
+    <Controller
       name={name}
-      initialValue={field.value}
-      normalize={handleNormalize}
-      validateStatus={fieldState.invalid ? 'error' : undefined}
-      help={fieldState.error?.message ?? help}
-    >
-      {children}
-    </AntdForm.Item>
+      control={control}
+      render={({ field }) => (
+        <AntdForm.Item
+          {...props}
+          validateStatus={fieldState.invalid ? 'error' : undefined}
+          help={fieldState.error?.message ?? help}
+        >
+          {Children.map(
+            children,
+            (child) =>
+              isValidElement(child) &&
+              cloneElement(child, {
```

Comment:
According to the latest react docs. [isValidElement](https://react.dev/reference/react/isValidElement) and [cloneElement](https://react.dev/reference/react/cloneElement) seems like legacy APIs. Are there any other alternatives?

Reply:
I had a look some [alternatives](https://react.dev/reference/react/cloneElement#alternatives) to [cloneElement](https://react.dev/reference/react/cloneElement) but I can't really see them working with the current API which is passing antd input components (e.g. `<Input />`, `<Select />`) directly as a child to that custom `<FormItem />`.

Reply:
thx, I'll try to handle this.